lib/ddtrace/tracer.rb

```diff
@@ -269,11 +269,10 @@ def trace(name, options = {})
         begin
           begin
             span = start_span(name, options)
-          # rubocop:disable Lint/UselessAssignment
           rescue StandardError => e
-            Datadog.logger.debug('Failed to start span: #{e}')
+            Datadog.logger.debug("Failed to start span: #{e}")
           ensure
-            return_value = yield(span)
+            return_value = yield(span) if span || e.is_a?(StandardError)
```

Comment:
not sure I follow this nested `begin`/`rescue` logic very well, maybe we need some comments? or a change in code structure to make L269:290 more clear?

Reply:
Documented!